subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -1090,6 +1141,17 @@ int aws_iot_init(const struct aws_iot_config *const config,
 		return err;
 	}
 
+	if (IS_ENABLED(CONFIG_AWS_IOT_BROKER_HOST_NAME_APP)) {
+		err = aws_iot_set_host_name(config->host_name, config->host_name_len);
+	} else {
+		err = aws_iot_set_host_name(NULL, 0);
```

Comment:
And provided `CONFIG_AWS_IOT_BROKER_HOST_NAME` as a parameter here:
`aws_iot_set_host_name(CONFIG_AWS_IOT_BROKER_HOST_NAME, strlen(CONFIG_AWS_IOT_BROKER_HOST_NAME))`

Reply:
It seems the compiler is not able to find `CONFIG_AWS_IOT_BROKER_HOST_NAME` in this case when `CONFIG_AWS_IOT_BROKER_HOST_NAME_APP` is enabled. I think because it has the property `depends on !AWS_IOT_BROKER_HOST_NAME_APP`. Shall I remove the `depends on` property, change it to `#ifdef` or leave it as is?

Reply:
I'd make it an `#ifdef`.